workflow/rules/authentic.smk

```diff
@@ -0,0 +1,186 @@
+# the checkpoint that shall trigger re-evaluation of the DAG
+checkpoint Extract_TaxIDs:
+    input:
+        pathogens="results/KRAKENUNIQ/{sample}/taxID.pathogens",
+    output:
+        dir=directory("results/AUTHENTICATION/{sample}"),
+    log:
+        "logs/EXTRACT_TAXIDS/{sample}.log",
+    shell:
+        "mkdir -p {output.dir}; "
+        "while read taxid; do mkdir {output.dir}/$taxid; done<{input.pathogens}"
+
+
+rule aggregate:
+    input:
+        aggregate_PMD,
+        aggregate_plots,
+        aggregate_post,
+    output:
+        "results/AUTHENTICATION/.{sample}_done",
+    log:
+        "logs/AGGREGATE/{sample}.log",
+    shell:
+        "touch {output}; "
+
+
+rule Make_Node_List:
+    input:
+        dir="results/AUTHENTICATION/{sample}/{taxid}/",
+    output:
+        node_list="results/AUTHENTICATION/{sample}/{taxid}/node_list.txt",
+    params:
+        tax_db=config["krakenuniq_db"],
+    log:
+        "logs/MAKE_NODE_LIST/{sample}_{taxid}.log",
+    shell:
+        "awk -v var={wildcards.taxid} '{{ if($1==var) print $0 }}' {params.tax_db}/taxDB | cut -f3 > {output.node_list}"
+
+
+rule Malt_Extract:
+    input:
+        rma6="results/MALT/{sample}.trimmed.rma6",
+        node_list="results/AUTHENTICATION/{sample}/{taxid}/node_list.txt",
+    output:
+        extract=directory(
+            "results/AUTHENTICATION/{sample}/{taxid}/{sample}.trimmed.rma6_MaltExtract_output"
+        ),
+    params:
+        ncbi_db=config["ncbi_db"],
+    threads: 4
+    log:
+        "logs/MALT_EXTRACT/{sample}_{taxid}.log",
+    conda:
+        "../envs/malt.yaml"
+    envmodules:
+        *config["envmodules"]["malt"],
+    message:
+        "RUNNING MALT EXTRACT FOR SAMPLE {input.rma6}"
+    shell:
+        "time MaltExtract -i {input.rma6} -f def_anc -o {output.extract} --reads --threads {threads} --matches --minPI 85.0 --maxReadLength 0 --minComp 0.0 --meganSummary -r {params.ncbi_db} -t {input.node_list} -v"
```

Comment:
Redirect output to log; currently all output goes to terminal

Reply:
This actually goes for other rules too, but this was the one with the most verbose output; Post_Processing was another